crates/trigger/src/lib.rs

```diff
@@ -113,18 +114,27 @@ impl<Executor: TriggerExecutor> TriggerExecutorBuilder<Executor> {
                 builder.add_host_component(outbound_redis::OutboundRedisComponent)?;
                 builder.add_host_component(outbound_pg::OutboundPg::default())?;
                 builder.add_host_component(outbound_mysql::OutboundMysql::default())?;
-                self.loader.add_dynamic_host_component(
-                    &mut builder,
-                    runtime_config::key_value::build_key_value_component(&runtime_config)?,
-                )?;
-                self.loader.add_dynamic_host_component(
-                    &mut builder,
-                    outbound_http::OutboundHttpComponent,
-                )?;
-                self.loader.add_dynamic_host_component(
-                    &mut builder,
-                    spin_config::ConfigHostComponent::new(runtime_config.config_providers()),
-                )?;
+                self.loader
+                    .add_dynamic_host_component(
+                        &mut builder,
+                        runtime_config::key_value::build_key_value_component(&runtime_config)?,
+                        init_data.kv,
```

Comment:
> I was kind of thinking it would be nice to have the initialisation login in the component itself

Sure!

```suggestion
                let component = runtime_config::key_value::build_key_value_component(&runtime_config)?;
                component.set_initial_values(init_data.kv)?;
                self.loader
                    .add_dynamic_host_component(
                        &mut builder,
                        component
```

```rust
impl KeyValueComponent {
  pub fn set_intitial_values ...
}
```

Reply:
...or some variation thereof... maybe the `set_initial_values` call would be better off inside `build_key_value_component` :shrug: